pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -59,13 +74,39 @@ func ValidateKubeSchedulerConfiguration(cc *config.KubeSchedulerConfiguration) f
 	return allErrs
 }
 
-// ValidateKubeSchedulerLeaderElectionConfiguration ensures validation of the KubeSchedulerLeaderElectionConfiguration struct
-func ValidateKubeSchedulerLeaderElectionConfiguration(cc *config.KubeSchedulerLeaderElectionConfiguration, fldPath *field.Path) field.ErrorList {
+func validateKubeSchedulerProfile(path *field.Path, profile *config.KubeSchedulerProfile) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(profile.SchedulerName) == 0 {
+		allErrs = append(allErrs, field.Required(path.Child("schedulerName"), ""))
+	}
+	return allErrs
+}
+
+func validateCommonQueueSort(path *field.Path, profiles []config.KubeSchedulerProfile) field.ErrorList {
+	allErrs := field.ErrorList{}
+	var canon *config.PluginSet
+	if profiles[0].Plugins != nil {
+		canon = profiles[0].Plugins.QueueSort
```

Comment:
Are the following cases valid?

- some profiles are missing queuesort plugin, however, queuesort plugin is at least specified in one profile
- the profile(s) with queuesort plugin specified is(are) not at `profiles[0]`

Reply:
Ideally, this validation logic shouldn't know about valid plugin names or what's the default plugin. That means it can't deduce if two configurations are equivalent. So, we decided that the queuesort plugin specification must be **strictly** equal. Then, the cases you describe are considered invalid.